http-server/src/main/java/io/micronaut/http/server/RouteExecutor.java

```diff
@@ -667,11 +676,126 @@ private Mono<MutableHttpResponse<?>> processPublisherBody(PropagatedContext prop
+
+        private boolean bufferWritten = false;
+
+        private FluxSink<Object> bodySink;
+
+        private boolean subscribed = false;
+
+        private Object bufferedFirstValue = null;
+
+        private Throwable bufferedImmediateError = null;
+
+        private final StreamSubscriber subscriber = new StreamSubscriber();
+
+        private boolean isComplete;
+
+        private boolean isEmpty;
+
+        private StreamingDataProcessor(Flux<Object> bodyPublisher) {
+            this.bodyPublisher = bodyPublisher;
+        }
+
+        @SuppressWarnings("unchecked")
+        private CorePublisher<MutableHttpResponse<?>> chunkedResponsePublisher(PropagatedContext propagatedContext,
+                                                                               HttpRequest<?> request,
+                                                                               MutableHttpResponse<?> response,
+                                                                               RouteInfo<?> routeInfo,
+                                                                               MediaType mediaType) {
+            Flux<Object> outputStream = Flux.push(this);
+
+            response
+                .header(HttpHeaders.TRANSFER_ENCODING, "chunked")
+                .header(HttpHeaders.CONTENT_TYPE, mediaType)
+                .body(ReactivePropagation.propagate(propagatedContext, outputStream));
+
+            Mono<? extends MutableHttpResponse<?>> responsePublisher = Mono.just(response)
+                .contextWrite(context -> ReactorPropagation.addPropagatedContext(context, propagatedContext).put(ServerRequestContext.KEY, request));
+
+            return (CorePublisher<MutableHttpResponse<?>>) outputStream.take(1)
+                .switchIfEmpty(responsePublisher)
+                .onErrorReturn(throwable -> findErrorRoute(throwable, routeInfo.getDeclaringType(), request) == null, response)
+                .flatMap(data -> responsePublisher)
+                .contextWrite(cv -> ReactorPropagation.addPropagatedContext(cv, propagatedContext).put(ServerRequestContext.KEY, request));
+        }
+
+        @Override
+        public void accept(FluxSink<Object> bodySink) {
+            this.bodySink = bodySink;
+            if (!subscribed) {
+                subscribed = true;
+                bodyPublisher.subscribe(subscriber);
+            } else if (!firstRequest && !bufferWritten) {
+                if (bufferedImmediateError != null) {
+                    bodySink.error(bufferedImmediateError);
+                } if (isEmpty){
+                    bodySink.complete();
+                } else {
+                    bodySink.next(bufferedFirstValue);
+                    if (isComplete) {
+                        bodySink.complete();
+                    }
+                }
+                bufferWritten = true;
+                bufferedFirstValue = null;
+                subscriber.request(Long.MAX_VALUE);
```

Comment:
we can't do that

Reply:
Yes, this was a mistake. I've replace this line with an onRequest callback on the sink that will still respect demand.

Reply:
it can still lose requests when the subscription happens after the downstream request